addons/website_sale/views/product_views.xml

```diff
@@ -111,6 +109,30 @@
         </field>
     </record>
 
+    <record id="product_template_view_form_inherit" model="ir.ui.view">
+        <field name="name">product.template.website.form.inherit</field>
+        <field name="model">product.template</field>
+        <field name="inherit_id" ref="product.product_template_only_form_view"/>
+        <field name="arch" type="xml">
+            <field name="public_categ_ids" position="after">
+                <field name="alternative_product_ids" widget="many2many_tags" domain="[('id', '!=', active_id), '|', ('company_id', '=', company_id), ('company_id', '=', False)]"/>
+                <field name="accessory_product_ids" widget="many2many_tags" domain="[('product_tmpl_id', '!=', active_id), '|', ('company_id', '=', company_id), ('company_id', '=', False)]"/>
```

Comment:
Main issue was to prevent multi company crashes.
Why bothering with preventing itself to be shown through adding `active_id` checks? I get the improvement but I think it is overkill in this case (stable).

I think if you just add the `('company_id', '=', company_id), ('company_id', '=', False)` domain onto both original fields without moving the fields this will fix the multi company issue.

Plus it will be possible to merge in stable, which is not the case with this PR.

Reply:
Done!